lib/chef/knife/vsphere_vm_snapshot.rb

```diff
@@ -67,14 +73,22 @@ def run
     end
 
     if config[:create_new_snapshot]
-      vm.CreateSnapshot_Task(name: config[:create_new_snapshot], description: '', memory: false, quiesce: false)
+	if config[:wait]
+      vm.CreateSnapshot_Task(name: config[:create_new_snapshot], description: '', memory: false, quiesce: false).wait_for_completion
```

Comment:
i don't think that would work,doesn't seem to,but perhaps this would:

snapshot_task=snapshot.RemoveSnapshot_Task(:removeChildren => false)
snapshot_task=snapshot_task.wait_for_completion if config[:wait]
snapshot_task
is that better ?